go/exocom-mock/exocom-mock.go

```diff
@@ -37,14 +38,18 @@ func (exoCom *ExoComMock) Listen(port int) error {
 	return exoCom.server.ListenAndServe()
 }
 
-// messageHandler is called when the given socket receives a message.
-// It logs the received message.
-func (exoCom *ExoComMock) messageHandler(socket *websocket.Conn) {
-	message, err := exoCom.readMessage(socket)
-	if err != nil {
-		panic(err)
+// websocketHandler is called when the given socket connects
+func (exoCom *ExoComMock) websocketHandler(socket *websocket.Conn) {
+	for {
+		message, err := exoCom.readMessage(socket)
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			panic(err)
+		} else {
+			exoCom.ReceivedMessages = append(exoCom.ReceivedMessages, message)
+		}
```

Comment:
Previously we were only reading the first message. Took me a while to figure this out